include/asm/symbol.h

```diff
@@ -20,6 +20,10 @@
 
 #define MAXSYMLEN	255
 
+// Issue #1157: allow defining a local label outside its parent
+// global label's scope, such as `func3.hSpam` within an HRAM section
+#define ALLOW_LOCAL_LABEL_ANYWHERE 1
```

Comment:
If the change is approved, it would just be the new behavior going forward. No need to call out a specific issue number in the comments and gate old unused code behind this constant.

Reply:
I guess I've developed bad habits at my day job, where I'm expected to gate newly requested behaviors behind build-time switches that my coworkers will tweak later as they balance the game. I'll hardcode the new behavior in the next revision.

Reply:
Sometimes a switch is okay, because users may not want the new behavior. For instance, there's a switch to reenable `ld` → `ldh` optimization, because users might want that. But those switches are _runtime_ switches, so you can use them when you use the program.

And in this case, such a switch is pointless. The new behavior is a true superset of the old behavior, so it makes no sense! At most, you can add a warning back (disabled by default), if you feel like that would be useful.